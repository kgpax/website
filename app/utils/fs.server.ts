export { default } from 'fs/promises'
