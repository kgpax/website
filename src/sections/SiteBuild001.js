import { Page, Segment, Paragraph, Code, ExternalLink } from '../components';

const SiteBuild001 = () => (
  <Page title="Building the site : part one">
    <Segment heading="Motivation">
      <Paragraph>
        I'd been toying with the idea of having my own website for a while. I'm
        pretty sure I made it a new years resolution for 2021, although when the
        lockdown kicked in and home-schooling (a.k.a parenting-from-work) became
        a thing, I'd lost any appetite I had for giving myself extra
        responsibility.
      </Paragraph>
      <Paragraph>
        I wasn't sure what I would put on a site... in fact at the time of
        writing this, I'm still not sure. I'm a bit of an oddball in that I'm
        self confident whilst also being quite shy and humble, and I find it
        hard to put myself or my ideas out there in such a public space. I'd
        been given suggestions about making a blog, or a portfolio site, or even
        just having a space to do some side projects and little fun hacks; all
        of which are good ideas.
      </Paragraph>
      <Paragraph>
        I figured that for my opening gambit, I could write about my experience
        of putting this site together, and share my thought processes and
        decisions along the way.
      </Paragraph>
      <Paragraph>
        Pro tip: I can ramble. If you're more a TL;DR person, this might not be
        your bag, but if you fancy embarking on my voyage of discovery, then
        strap in.
      </Paragraph>
    </Segment>

    <Segment heading="Picking the tech stack">
      <Paragraph>
        I'm a full stack developer by trade, or at least historically I am. My
        passion is for front-end development and my current go-to front-end
        library is{' '}
        <ExternalLink href="https://reactjs.org/">React</ExternalLink>. Despite
        my current job being in a front-end team, I haven't actually gotten the
        opportunity to do a great deal of front-end work and keep up with modern
        front-end trends as much as I'd like. I toyed with the idea of using
        this website as an opportunity to skill up on{' '}
        <ExternalLink href="https://nextjs.org/">NextJS</ExternalLink> for
        universal rendering, or even go down the static site generator route
        with{' '}
        <ExternalLink href="https://www.gatsbyjs.com/">Gatsby</ExternalLink>, or{' '}
        <ExternalLink href="https://www.11ty.dev/">Eleventy</ExternalLink>, or
        perhaps one of the many others options which I'm not really caught up
        with because I've been criminally out of touch with front-end tech
        stacks.
      </Paragraph>
      <Paragraph>
        In the end, I've opted to reduce the number of learning curves and just
        go the way of using <Code>create-react-app</Code>. It's a comfort zone
        and one which allows me to focus on the early build out of the site
        without getting bamboozled and ultimately disincentivised right at the
        start. Heck, check out the{' '}
        <ExternalLink href="https://github.com/kgpax/website/tree/sitebuild-001">
          source code at the time of this version
        </ExternalLink>{' '}
        and you'll see that I'm currently building these pages using components
        rather than using markdown which builds to HTML. I think that's
        old-school now, right?
      </Paragraph>
      <Paragraph>
        I'd toyed with the idea of using{' '}
        <ExternalLink href="https://svelte.dev/">Svelte</ExternalLink> instead
        of React because I like the idea of compiling away the framework and
        having plain-ol` javascript without all of the extra weight; however, my
        ultimate aim would be to not be so dependant on javascript at all and
        instead look to have the content as static as possible for better
        performance and SEO; all things which I have not had much in the way of
        personal experience in. As such, I'll probably go down the route of
        static site generators and migrate all of this code over as part of a
        learning process.
      </Paragraph>
    </Segment>

    <Segment heading="The design">
      <Paragraph>
        I'm quite a visually driven person. I like to think I have a good eye
        for design, but that doesn't necessarily translate to being{' '}
        <em>good at</em> design.
      </Paragraph>
      <Paragraph>
        I've opted to start fairly minimalist, whilst putting at least some
        thought into colour, typography and spacing. I plan to scour the web for
        inspiration on how I can enhance the aesthetics of this website, and use
        it as an opportunity to flex some of those UI and UX muscles which have
        been wasting away for too long.
      </Paragraph>
    </Segment>

    <Segment heading="My first go-live">
      <Paragraph>
        I purchased the <Code>kevinpaxton.com</Code> domain years ago, around
        the time that Google launched its <Code>.dev</Code> top level domains. I
        didn't have a concrete plan at the time, I just wanted to see if the
        domain was available for my name and get it whilst I could.
      </Paragraph>
      <Paragraph>
        Since then, it has just sat there, squatting; I've renewed it twice
        since then and only remembered I had it when I got the renewal emails.
        In fact, it's been so long out of my mind that when I deployed the first
        version of this site,{' '}
        <ExternalLink href="https://github.com/kgpax/website/commit/f855e83657d20770641fb00cea30c795e8693eb8">
          I got the domain wrong
        </ExternalLink>
        , thinking I'd purchased <Code>kevinpaxton.dev</Code> instead of
        securing the <Code>.com</Code> domain.
      </Paragraph>
      <Paragraph>
        I was recommended using{' '}
        <ExternalLink href="https://app.netlify.com/">Netlify</ExternalLink> as
        a quick and easy way to host my site by a{' '}
        <ExternalLink href="https://twitter.com/CarolSaysThings">
          friend of mine
        </ExternalLink>{' '}
        and sure enough the process was really streamlined and really quite
        cool. I loved how once it got your permission to go to town on your
        repo, it just set everything up for you. I'm familiar with pipelines and
        whatnot integrating with GitHub, but I was still pleasantly surprised at
        how hands-off the process was and how when it was completed, it Just
        Works&trade;. I had to do some Google-Fu to get my Google domain
        pointing to my Netlify app, but that's mainly because I didn't RTFM and
        tried to go it alone with my painfully limited DNS knowledge, but in the
        end - here we are.
      </Paragraph>
      <Paragraph>
        The start has been made. Now I've got to keep it up.
      </Paragraph>
    </Segment>

    <Segment heading="Next steps">
      <Paragraph>
        I'm going to try an make this a regular section to my updates, where I
        list bits I want to go after next. I won't hold myself to getting
        everything done, but it will serve as an interesting retrospective for
        my journey.
      </Paragraph>
      <Paragraph>
        Here's what I hope to put up next time:
        <ul>
          <li>
            Put some motion on the landing page - I love a bit of subtle
            animation, me
          </li>
          <li>
            Work on the page backgrounds - change those straight lines to
            curves?
          </li>
          <li>Favicon and early browser and device testing</li>
          <li>Think about the navigation structure of the site</li>
        </ul>
        Stretch goals:
        <ul>
          <li>
            Get some illustrations and imagery in place - this is far too text
            heavy, I want to fix that
          </li>
        </ul>
      </Paragraph>
    </Segment>
  </Page>
);

export default SiteBuild001;
