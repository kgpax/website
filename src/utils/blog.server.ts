import glob from 'fast-glob'
import {
  BlogData,
  BlogEntry,
  BlogSeriesEntry,
  EntryMetaData,
} from '~/types/blog'

export async function getBlogEntries(): Promise<BlogEntry[]> {
  const postPaths = await glob('src/pages/blog/**/*.mdx')

  const posts = await Promise.all(
    postPaths.map(async path => {
      const filePath = path.replace(/^src\/pages/, '')
      const fileName = path.split('/').reverse()[0].replace(/\.mdx/, '')
      const slug = filePath.replace(/\.mdx/, '')
      const series = filePath
        .substring(0, filePath.lastIndexOf('/'))
        .replace(/^\/blog\//, '')
      const { meta }: { meta: EntryMetaData } = await import(
        `../pages${filePath}`
      )

      return {
        ...meta,
        filePath,
        fileName,
        series,
        slug,
        isPublished: !!meta.date,
      } as BlogData
    }),
  )

  const sortedPosts = posts.sort((a, b) =>
    (a.date ?? '9999-99-99').localeCompare(b.date ?? '9999-99-99'),
  )

  const entries: BlogEntry[] = []

  for (const post of sortedPosts) {
    if (!post.series) {
      entries.push({
        type: 'post',
        ...post,
      })
    } else {
      if (post.series) {
        let series = entries.find(
          entry => entry.series === post.series,
        ) as BlogSeriesEntry
        if (!series) {
          series = {
            type: 'series',
            ...post,
            posts: [],
          }
          entries.push(series)
        }
        if (post.fileName !== '_meta') {
          series.posts.push({ type: 'post', ...post })
        }
      } else {
      }
    }
  }

  return entries
}
