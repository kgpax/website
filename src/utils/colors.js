const pairs = [
  ['#D7F2BA', '#BDE4A8'],
  ['#FFF200', '#FFD300'],
  ['#97D2FB', '#83BCFF'],
  ['#FFBE85', '#FF9B42'],
];

export const pickColorPair = () =>
  pairs[Math.floor(Math.random() * pairs.length)];
