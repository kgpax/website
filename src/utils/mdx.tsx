import { MDXComponents } from 'mdx/types'
import { BlogPage, Heading, Link, MdxCode } from '~/components'

export const components: MDXComponents = {
  wrapper: props => {
    console.log(props)
    return <BlogPage {...props}>{props.children}</BlogPage>
  },
  h2: Heading,
  a: Link,
  code: MdxCode,
}
