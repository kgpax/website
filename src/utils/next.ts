import { GetStaticPropsResult } from 'next'

export function next404<T>(): GetStaticPropsResult<T> {
  return { notFound: true }
}
