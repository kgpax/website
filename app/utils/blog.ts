import parseFrontMatter from 'front-matter'
import { BlogInfo, BlogMetadata, BlogMetadataAndUrl } from '~/types/blog'
import fs from '~/utils/fs.server'
import { bundleMDX } from '~/utils/mdx.server'
import process from '~/utils/process.server'

const blogContentPath = (filename: string = '') =>
  `${process.cwd()}/app/content/blog/${filename}`

export async function getPost(slug: string): Promise<BlogInfo | null> {
  const source = await fs
    .readFile(blogContentPath(`${slug}.mdx`), 'utf-8')
    .catch(() => null)

  if (source === null) return null

  const post = await bundleMDX({
    source,
    mdxOptions(options) {
      options.remarkPlugins = [...(options.remarkPlugins ?? [])]
      options.rehypePlugins = [...(options.rehypePlugins ?? [])]
      return options
    },
    esbuildOptions(options) {
      options.platform = 'node'
      return options
    },
    cwd: blogContentPath(),
  })

  return {
    code: post.code,
    url: `/blog/${slug}`,
    ...(post.frontmatter as BlogMetadata),
  }
}

export async function getPosts(): Promise<BlogMetadataAndUrl[]> {
  const postsPath = await fs.readdir(blogContentPath(), {
    withFileTypes: true,
  })

  const posts = await Promise.all(
    postsPath.map(async entry => {
      const file = await fs.readFile(blogContentPath(entry.name))
      const { attributes } = parseFrontMatter<BlogMetadata>(file.toString())

      return {
        url: `/blog/${entry.name.replace(/\.mdx/, '')}`,
        ...attributes,
      } as BlogMetadataAndUrl
    }),
  )

  return posts
}
