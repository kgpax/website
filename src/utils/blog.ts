import { BlogSeriesEntry } from '~/types/blog'

export function getBlogPostSeries(
  series: BlogSeriesEntry[],
  id: string,
): BlogSeriesEntry | null {
  return series.find(x => x.posts.some(y => y.id === id)) ?? null
}
