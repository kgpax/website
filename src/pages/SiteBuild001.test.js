import { performSnapshotTest } from '../../test/utils';
import SiteBuild001 from './SiteBuild001';

test('should match snapshot', () => {
  performSnapshotTest(<SiteBuild001 />);
});
