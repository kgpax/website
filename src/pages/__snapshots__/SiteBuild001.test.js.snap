// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  class="sc-hKFxyN gcDctk"
>
  <svg
    class="sc-dlnjwi eZHRFH sc-eCApnc hcAKMR"
    height="320"
    preserveAspectRatio="none"
    viewBox="0 0 1440 320"
    width="100%"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      class="sc-bdnxRM eMaIpy"
      d="M0,224L48,192C96,160,192,96,288,90.7C384,85,480,139,576,138.7C672,139,768,85,864,101.3C960,117,1056,203,1152,224C1248,245,1344,203,1392,181.3L1440,160L1440,320L1392,320C1344,320,1248,320,1152,320C1056,320,960,320,864,320C768,320,672,320,576,320C480,320,384,320,288,320C192,320,96,320,48,320L0,320Z"
      fill="#FFFFFF"
      fill-opacity="0.35"
    />
    <path
      class="sc-gtsrHT dUsYnb"
      d="M0,224L48,192C96,160,192,96,288,90.7C384,85,480,139,576,138.7C672,139,768,85,864,101.3C960,117,1056,203,1152,224C1248,245,1344,203,1392,181.3L1440,160L1440,320L1392,320C1344,320,1248,320,1152,320C1056,320,960,320,864,320C768,320,672,320,576,320C480,320,384,320,288,320C192,320,96,320,48,320L0,320Z"
      fill="#FFFFFF"
      fill-opacity="0.35"
    >
      <animate
        attributeName="d"
        calcMode="spline"
        dur="30s"
        keySplines="0.45, 0, 0.55, 1;0.45, 0, 0.55, 1;0.45, 0, 0.55, 1"
        keyTimes="0;0.2;0.8;1"
        repeatCount="indefinite"
        values="M0,224L48,192C96,160,192,96,288,90.7C384,85,480,139,576,138.7C672,139,768,85,864,101.3C960,117,1056,203,1152,224C1248,245,1344,203,1392,181.3L1440,160L1440,320L1392,320C1344,320,1248,320,1152,320C1056,320,960,320,864,320C768,320,672,320,576,320C480,320,384,320,288,320C192,320,96,320,48,320L0,320Z;M0,128L48,154.7C96,181,192,235,288,234.7C384,235,480,181,576,138.7C672,96,768,64,864,48C960,32,1056,32,1152,48C1248,64,1344,96,1392,112L1440,128L1440,320L1392,320C1344,320,1248,320,1152,320C1056,320,960,320,864,320C768,320,672,320,576,320C480,320,384,320,288,320C192,320,96,320,48,320L0,320Z;M0,288L48,261.3C96,235,192,181,288,133.3C384,85,480,43,576,58.7C672,75,768,149,864,170.7C960,192,1056,160,1152,170.7C1248,181,1344,235,1392,261.3L1440,288L1440,320L1392,320C1344,320,1248,320,1152,320C1056,320,960,320,864,320C768,320,672,320,576,320C480,320,384,320,288,320C192,320,96,320,48,320L0,320Z;M0,224L48,192C96,160,192,96,288,90.7C384,85,480,139,576,138.7C672,139,768,85,864,101.3C960,117,1056,203,1152,224C1248,245,1344,203,1392,181.3L1440,160L1440,320L1392,320C1344,320,1248,320,1152,320C1056,320,960,320,864,320C768,320,672,320,576,320C480,320,384,320,288,320C192,320,96,320,48,320L0,320Z"
      />
    </path>
    <path
      class="sc-bdnxRM eMaIpy"
      d="M0,288L48,261.3C96,235,192,181,288,133.3C384,85,480,43,576,58.7C672,75,768,149,864,170.7C960,192,1056,160,1152,170.7C1248,181,1344,235,1392,261.3L1440,288L1440,320L1392,320C1344,320,1248,320,1152,320C1056,320,960,320,864,320C768,320,672,320,576,320C480,320,384,320,288,320C192,320,96,320,48,320L0,320Z"
      fill="#FFFFFF"
      fill-opacity="0.5"
    />
    <path
      class="sc-gtsrHT dUsYnb"
      d="M0,288L48,261.3C96,235,192,181,288,133.3C384,85,480,43,576,58.7C672,75,768,149,864,170.7C960,192,1056,160,1152,170.7C1248,181,1344,235,1392,261.3L1440,288L1440,320L1392,320C1344,320,1248,320,1152,320C1056,320,960,320,864,320C768,320,672,320,576,320C480,320,384,320,288,320C192,320,96,320,48,320L0,320Z"
      fill="#FFFFFF"
      fill-opacity="0.5"
    >
      <animate
        attributeName="d"
        calcMode="spline"
        dur="25s"
        keySplines="0.45, 0, 0.55, 1;0.45, 0, 0.55, 1;0.45, 0, 0.55, 1"
        keyTimes="0;0.3;0.7;1"
        repeatCount="indefinite"
        values="M0,288L48,261.3C96,235,192,181,288,133.3C384,85,480,43,576,58.7C672,75,768,149,864,170.7C960,192,1056,160,1152,170.7C1248,181,1344,235,1392,261.3L1440,288L1440,320L1392,320C1344,320,1248,320,1152,320C1056,320,960,320,864,320C768,320,672,320,576,320C480,320,384,320,288,320C192,320,96,320,48,320L0,320Z;M0,224L48,192C96,160,192,96,288,90.7C384,85,480,139,576,138.7C672,139,768,85,864,101.3C960,117,1056,203,1152,224C1248,245,1344,203,1392,181.3L1440,160L1440,320L1392,320C1344,320,1248,320,1152,320C1056,320,960,320,864,320C768,320,672,320,576,320C480,320,384,320,288,320C192,320,96,320,48,320L0,320Z;M0,128L48,154.7C96,181,192,235,288,234.7C384,235,480,181,576,138.7C672,96,768,64,864,48C960,32,1056,32,1152,48C1248,64,1344,96,1392,112L1440,128L1440,320L1392,320C1344,320,1248,320,1152,320C1056,320,960,320,864,320C768,320,672,320,576,320C480,320,384,320,288,320C192,320,96,320,48,320L0,320Z;M0,288L48,261.3C96,235,192,181,288,133.3C384,85,480,43,576,58.7C672,75,768,149,864,170.7C960,192,1056,160,1152,170.7C1248,181,1344,235,1392,261.3L1440,288L1440,320L1392,320C1344,320,1248,320,1152,320C1056,320,960,320,864,320C768,320,672,320,576,320C480,320,384,320,288,320C192,320,96,320,48,320L0,320Z"
      />
    </path>
    <path
      class="sc-bdnxRM eMaIpy"
      d="M0,128L48,154.7C96,181,192,235,288,234.7C384,235,480,181,576,138.7C672,96,768,64,864,48C960,32,1056,32,1152,48C1248,64,1344,96,1392,112L1440,128L1440,320L1392,320C1344,320,1248,320,1152,320C1056,320,960,320,864,320C768,320,672,320,576,320C480,320,384,320,288,320C192,320,96,320,48,320L0,320Z"
      fill="#FFFFFF"
      fill-opacity="1"
    />
    <path
      class="sc-gtsrHT dUsYnb"
      d="M0,128L48,154.7C96,181,192,235,288,234.7C384,235,480,181,576,138.7C672,96,768,64,864,48C960,32,1056,32,1152,48C1248,64,1344,96,1392,112L1440,128L1440,320L1392,320C1344,320,1248,320,1152,320C1056,320,960,320,864,320C768,320,672,320,576,320C480,320,384,320,288,320C192,320,96,320,48,320L0,320Z"
      fill="#FFFFFF"
      fill-opacity="1"
    >
      <animate
        attributeName="d"
        calcMode="spline"
        dur="35s"
        keySplines="0.45, 0, 0.55, 1;0.45, 0, 0.55, 1;0.45, 0, 0.55, 1"
        keyTimes="0;0.4;0.6;1"
        repeatCount="indefinite"
        values="M0,128L48,154.7C96,181,192,235,288,234.7C384,235,480,181,576,138.7C672,96,768,64,864,48C960,32,1056,32,1152,48C1248,64,1344,96,1392,112L1440,128L1440,320L1392,320C1344,320,1248,320,1152,320C1056,320,960,320,864,320C768,320,672,320,576,320C480,320,384,320,288,320C192,320,96,320,48,320L0,320Z;M0,288L48,261.3C96,235,192,181,288,133.3C384,85,480,43,576,58.7C672,75,768,149,864,170.7C960,192,1056,160,1152,170.7C1248,181,1344,235,1392,261.3L1440,288L1440,320L1392,320C1344,320,1248,320,1152,320C1056,320,960,320,864,320C768,320,672,320,576,320C480,320,384,320,288,320C192,320,96,320,48,320L0,320Z;M0,224L48,192C96,160,192,96,288,90.7C384,85,480,139,576,138.7C672,139,768,85,864,101.3C960,117,1056,203,1152,224C1248,245,1344,203,1392,181.3L1440,160L1440,320L1392,320C1344,320,1248,320,1152,320C1056,320,960,320,864,320C768,320,672,320,576,320C480,320,384,320,288,320C192,320,96,320,48,320L0,320Z;M0,128L48,154.7C96,181,192,235,288,234.7C384,235,480,181,576,138.7C672,96,768,64,864,48C960,32,1056,32,1152,48C1248,64,1344,96,1392,112L1440,128L1440,320L1392,320C1344,320,1248,320,1152,320C1056,320,960,320,864,320C768,320,672,320,576,320C480,320,384,320,288,320C192,320,96,320,48,320L0,320Z"
      />
    </path>
  </svg>
  <div
    class="sc-jSFjdj iqrXij"
  >
    <section
      class="sc-bqGGPW cMYPGv"
    >
      <header
        class="sc-gKAaRy sc-jrsJWt eflIJP"
      >
        ↑
        <a
          class="sc-iwajpm fEDrup"
          href="/"
        >
          kevinpaxton.com
        </a>
      </header>
      <article
        class="sc-iCoGMd dSirmf sc-kEqXSa bYFfzU"
      >
        <time
          class="sc-dlMDgC hgbZqc"
        >
          Publish date TBC
        </time>
        <section
          class="sc-hBMUJo mTCpl"
          id="motivation"
        >
          <h2
            class="sc-ksluID bOubR"
          >
            Motivation
          </h2>
          <p
            class="sc-fnVZcZ bWseFA"
          >
            I'd been toying with the idea of having my own website for a while. I'm pretty sure I made it a new years resolution for 2021, although when the lockdown kicked in and home-schooling (a.k.a parenting-from-work) became a thing, I'd lost any appetite I had for giving myself extra responsibility.
          </p>
          <p
            class="sc-fnVZcZ bWseFA"
          >
            I wasn't sure what I would put on a site... in fact at the time of writing this I'm still not sure. I'm a bit of an odd person — some would say stop the sentence there — in that I'm confident and have a lot of self-belief whilst simultaneously being quite shy, humble and self-effacing about putting myself or the work I produce out there in such a public space; I'm not entirely sure
             
            <a
              class="sc-cxNHIi fsKHxE"
              href="https://en.wikipedia.org/wiki/Impostor_syndrome"
              rel="nofollow noopener noreferrer"
              target="_blank"
            >
              why this is
            </a>
            , but putting this website out there is a way to try and fight those demons. I'd been given suggestions about making a blog, or a portfolio site, or even just having a space to do some side projects and little fun hacks; all of which are good ideas, and all of which might make their way into it at some point.
          </p>
          <p
            class="sc-fnVZcZ bWseFA"
          >
            I figured that for my opening gambit, I could write about my experience of putting this site together, and share my thought processes and decisions along the way.
          </p>
          <p
            class="sc-fnVZcZ bWseFA sc-hHEiqL bQzmHZ"
          >
            <img
              alt="Speech bubble, hinting at lots of talking"
              class="sc-fFSPTT"
              src="img/speech.svg"
            />
            <span>
              <strong>
                Pro tip:
              </strong>
               I can ramble. If you're more a TL;DR person, this might not be your bag, but if you fancy embarking on my voyage of discovery, then strap in.
            </span>
          </p>
        </section>
        <section
          class="sc-hBMUJo mTCpl"
          id="picking-the-tech-stack"
        >
          <h2
            class="sc-ksluID bOubR"
          >
            Picking the tech stack
          </h2>
          <p
            class="sc-fnVZcZ bWseFA"
          >
            I'm a full stack developer by trade, or at least historically I am. My passion is for front-end development and my current go-to front-end library is 
            <a
              class="sc-cxNHIi fsKHxE"
              href="https://reactjs.org/"
              rel="nofollow noopener noreferrer"
              target="_blank"
            >
              React
            </a>
            . Despite my current job being in a front-end team, I haven't actually gotten the opportunity to do a great deal of front-end work and keep up with modern front-end trends as much as I'd like. I toyed with the idea of using this website as an opportunity to skill up on
             
            <a
              class="sc-cxNHIi fsKHxE"
              href="https://nextjs.org/"
              rel="nofollow noopener noreferrer"
              target="_blank"
            >
              NextJS
            </a>
             for universal rendering, or even go down the static site generator route with
             
            <a
              class="sc-cxNHIi fsKHxE"
              href="https://www.gatsbyjs.com/"
              rel="nofollow noopener noreferrer"
              target="_blank"
            >
              Gatsby
            </a>
            , or
             
            <a
              class="sc-cxNHIi fsKHxE"
              href="https://www.11ty.dev/"
              rel="nofollow noopener noreferrer"
              target="_blank"
            >
              Eleventy
            </a>
            , or perhaps one of the many others options which I'm not really caught up with because I've been criminally out of touch with front-end tech stacks.
          </p>
          <p
            class="sc-fnVZcZ bWseFA"
          >
            In the end, I've opted to reduce the number of learning curves and just go the way of using 
            <code
              class="sc-kfYoZR ldJYzE"
            >
              create-react-app
            </code>
            . It's a comfort zone and one which allows me to focus on the early build out of the site without getting bamboozled and ultimately disincentivised right at the start. Heck, check out the
             
            <a
              class="sc-cxNHIi fsKHxE"
              href="https://github.com/kgpax/website/tree/sitebuild-001"
              rel="nofollow noopener noreferrer"
              target="_blank"
            >
              source code at the time of this version
            </a>
             
            and you'll see that I'm currently building these pages using components rather than using markdown which builds to HTML. I think that's old-school now, right?
          </p>
          <p
            class="sc-fnVZcZ bWseFA"
          >
            I'd toyed with the idea of using
             
            <a
              class="sc-cxNHIi fsKHxE"
              href="https://svelte.dev/"
              rel="nofollow noopener noreferrer"
              target="_blank"
            >
              Svelte
            </a>
             instead of React because I like the idea of compiling away the framework and having plain-ol\` javascript without all of the extra weight; however, my ultimate aim would be to not be so dependant on javascript at all and instead look to have the content as static as possible for better performance and SEO; all things which I have not had much in the way of personal experience in. As such, I'll probably go down the route of static site generators and migrate all of this code over as part of a learning process.
          </p>
        </section>
        <section
          class="sc-hBMUJo mTCpl"
          id="the-design"
        >
          <h2
            class="sc-ksluID bOubR"
          >
            The design
          </h2>
          <p
            class="sc-fnVZcZ bWseFA"
          >
            I'm quite a visually driven person. I like to think I have a good eye for design, but that doesn't necessarily translate to being
             
            <em>
              good at
            </em>
             design.
          </p>
          <p
            class="sc-fnVZcZ bWseFA"
          >
            I've opted to start fairly minimalist, whilst putting at least some thought into colour, typography and spacing. I plan to scour the web for inspiration on how I can enhance the aesthetics of this website, and use it as an opportunity to flex some of those UI and UX muscles which have been wasting away for too long.
          </p>
        </section>
        <section
          class="sc-hBMUJo mTCpl"
          id="my-first-go-live"
        >
          <h2
            class="sc-ksluID bOubR"
          >
            My first go-live
          </h2>
          <p
            class="sc-fnVZcZ bWseFA"
          >
            I purchased the 
            <code
              class="sc-kfYoZR ldJYzE"
            >
              kevinpaxton.com
            </code>
             domain years ago, around the time that Google launched its 
            <code
              class="sc-kfYoZR ldJYzE"
            >
              .dev
            </code>
             top level domains. I didn't have a concrete plan at the time, I just wanted to see if the domain was available for my name and get it whilst I could.
          </p>
          <p
            class="sc-fnVZcZ bWseFA"
          >
            Since then, it has just sat there, squatting; I've renewed it twice since then and only remembered I had it when I got the renewal emails. In fact, it's been so long out of my mind that when I deployed the first version of this site,
             
            <a
              class="sc-cxNHIi fsKHxE"
              href="https://github.com/kgpax/website/commit/f855e83657d20770641fb00cea30c795e8693eb8"
              rel="nofollow noopener noreferrer"
              target="_blank"
            >
              I got the domain wrong
            </a>
            , thinking I'd purchased 
            <code
              class="sc-kfYoZR ldJYzE"
            >
              kevinpaxton.dev
            </code>
             instead of securing the
             
            <code
              class="sc-kfYoZR ldJYzE"
            >
              .com
            </code>
             domain.
          </p>
          <p
            class="sc-fnVZcZ bWseFA"
          >
            I was recommended using 
            <a
              class="sc-cxNHIi fsKHxE"
              href="https://app.netlify.com/"
              rel="nofollow noopener noreferrer"
              target="_blank"
            >
              Netlify
            </a>
             
            as a quick and easy way to host my site by a
             
            <a
              class="sc-cxNHIi fsKHxE"
              href="https://twitter.com/CarolSaysThings"
              rel="nofollow noopener noreferrer"
              target="_blank"
            >
              friend of mine
            </a>
             and sure enough the process was really streamlined and really quite cool. I loved how once it got your permission to go to town on your repo, it just set everything up for you. I'm familiar with pipelines and whatnot integrating with GitHub, but I was still pleasantly surprised at how hands-off the process was and how when it was completed, it Just Works™. I had to do some Google-Fu to get my Google domain pointing to my Netlify app, but that's mainly because I didn't RTFM and tried to go it alone with my painfully limited DNS knowledge, but in the end - here we are.
          </p>
          <p
            class="sc-fnVZcZ bWseFA"
          >
            A start has been made. Now I've got to keep it up.
          </p>
        </section>
        <section
          class="sc-hBMUJo mTCpl"
          id="next-steps"
        >
          <h2
            class="sc-ksluID bOubR"
          >
            Next steps
          </h2>
          <p
            class="sc-fnVZcZ bWseFA"
          >
            I'm going to try and make this a regular section to my updates, where I list bits I want to go after next. I won't hold myself to getting everything done, but it will serve as an interesting retrospective for my journey.
          </p>
          <p
            class="sc-fnVZcZ bWseFA"
          >
            Here's what I hope to put up next time:
          </p>
          <ul>
            <li>
              Put some motion on the landing page - I love a bit of subtle animation, me
            </li>
            <li>
              Work on the page backgrounds - change those straight lines to curves?
            </li>
            <li>
              Favicon and early browser and device testing
            </li>
            <li>
              Think about the navigation structure of the site
            </li>
          </ul>
          <p
            class="sc-fnVZcZ bWseFA"
          >
            Stretch goals:
          </p>
          <ul>
            <li>
              Get some illustrations and imagery in place - this is far too text heavy, I want to fix that
            </li>
          </ul>
        </section>
      </article>
      <footer
        class="sc-fujyAs sc-iqAclL eoquod"
      >
        <span
          class="sc-pNWdM PnHWI"
        >
          © 
          2021
           
          <a
            class="sc-iwajpm fEDrup"
            href="/"
          >
            Kevin Paxton
          </a>
        </span>
      </footer>
    </section>
  </div>
</div>
`;
